utilities/scripts/python/diff.py

```diff
@@ -0,0 +1,156 @@
+from dataclasses import dataclass
+from typing import Dict
+
+
+@dataclass
+class DiffEntry:
+    """
+    A data class representing an entry in a difference collection.
+    Attributes:
+    val: any
+        The value of the entry.
+    add_or_remove_flag: bool
+        A flag indicating whether the entry should be removed (False) or added (True).
+    """
+
+    val: any
+    add_or_remove_flag: bool
+
+
+@dataclass
+class Diff:
+    """
+    Represents the difference between two dictionaries.
+    """
+    diff: dict
+
```

Comment:
add `expected` and `provided` and `strict` here in the class data.
Then in `__init__` assign them,
Then you should not pass them as parameters any more, but should refer to them with `self.expected` and so on/